Sources/MapboxCoreNavigation/Telemetry/EventStep.swift

```diff
@@ -0,0 +1,104 @@
+import Foundation
+import CoreLocation
+import MapboxNavigationNative
+
+struct EventStep: Equatable {
+    let distance: Double
+    let distanceRemaining: Double
+    let duration: Double
+    let durationRemaining: Double
+    let upcomingName: String
+    let upcomingType: String
+    let upcomingModifier: String
+    let upcomingInstruction: String
+    let previousName: String
+    let previousType: String
+    let previousModifier: String
+    let previousInstruction: String
+    /**
+     Initializes an event location consistent with the given location object.
+     */
+    init(_ step: Step) {
+        self.distance = step.distance
+        self.distanceRemaining = step.distanceRemaining
+        self.duration = step.duration
+        self.durationRemaining = step.durationRemaining
+        self.upcomingName = step.upcomingName
+        self.upcomingType = step.upcomingType
+        self.upcomingModifier = step.upcomingModifier
+        self.upcomingInstruction = step.upcomingInstruction
+        self.previousName = step.previousName
+        self.previousType = step.previousType
+        self.previousModifier = step.previousModifier
+        self.previousInstruction = step.previousInstruction
+    }
+}
+
+extension EventStep: Codable {
+    private enum CodingKeys: String, CodingKey {
+        case distance
+        case distanceRemaining
+        case duration
+        case durationRemaining
+        case upcomingName
+        case upcomingType
+        case upcomingModifier
+        case upcomingInstruction
+        case previousName
+        case previousType
+        case previousModifier
+        case previousInstruction
+    }
+
+    init(from decoder: Decoder) throws {
```

Comment:
I wonder why we need a custom implementation of Codable protocol, looks like the Swift compiler would generate the same by default

Reply:
Thanks for noticing. It is not needed, I forgot to remove it after the initial non-struct implementation